sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -679,7 +679,10 @@ object MapObjects {
       elementNullable: Boolean = true,
       customCollectionCls: Option[Class[_]] = None): MapObjects = {
     val loopVar = LambdaVariable("MapObject", elementType, elementNullable)
-    MapObjects(loopVar, function(loopVar), inputData, customCollectionCls)
+    // println(s"\n\n $function, $inputData, $elementType, $elementNullable, $customCollectionCls \n")
+    val fOfLV = if (function == null) loopVar else function(loopVar)
```

Comment:
Looks not correct fix as this only works if `function` is just "identity"? 

`UnresolvedMapObjects` was added long while ago, and for the transient `function`, seems it was added without test case. So I can just tell from the code comment, I guess that that `UnresolvedMapObjects` should not be resolved and evaluated on executors.

Reply:
Yea this is the core issue: we don't expect to serialize and send `UnresolvedMapObjects` to executors. Sometimes users may carry it if the RDD closure is not carefully written. We allow this mistake but we can't allow if users do want to process the `UnresolvedMapObjects` at executor side.

Reply:
> I guess that that UnresolvedMapObjects should not be resolved and evaluated on executors.

As of SPARK-27926 that has changed. This change is the source of the bug.

> this only works if function is just "identity"?

Yes. My main concern with this fix is determining whether "identity" will work in all cases, and if it will not, what else might be done.